### PR TITLE
Memory Only Storage Option

### PR DESCRIPTION
**Pull request by AliRKat**

### CODE REVIEW

**Comment by AliRKat** [comment]
[[NodeJS] Option to use memory only for storage](https://trello.com/c/3B9JMGQg/1943-nodejs-option-to-use-memory-only-for-storage)

File: lib/countly-bulk.js [hunk 1]
```diff
@@ -102,7 +106,10 @@ function CountlyBulk(conf) {
     conf.maxStackTraceLinesPerThread = conf.max_stack_trace_lines_per_thread || maxStackTraceLinesPerThread;
     conf.maxStackTraceLineLength = conf.max_stack_trace_line_length || maxStackTraceLineLength;
 
-    CountlyStorage.setStoragePath(conf.storage_path, true, conf.persist_queue);
+    if (!conf.storage_type) {
```
Comment (turtledreams) [inline lib/countly-bulk.js:109]: so 2 lines above is an example of how we want to handle provided config options

File: test/tests_storage.js [hunk 2]
```diff
@@ -266,7 +267,41 @@ describe("Storage Tests", () => {
...
+    it("17- Setting storage method to memory only and checking storage path /no-init", (done) => {
+        storage.resetStorage();
+        storage.initStorage(null, cc.storageTypeEnums.MEMORY);
+        assert.equal(storage.getStoragePath(), undefined);
+        done();
+    });
+
+    it("18- Recording to storage with memory only storage /no-init", (done) => {
+        storage.resetStorage();
+        recordValuesToStorageAndValidate(null, cc.storageTypeEnums.MEMORY);
```
Comment (turtledreams) [inline test/tests_storage.js:304]: we need SDK running with different storage config options. not testing the module internally. so a real usecase, like initial tests above.

**Review by turtledreams — CHANGES_REQUESTED** [review]

**Review by turtledreams — CHANGES_REQUESTED** [review]
verify that persistent storage folder is empty in memory tests

File: lib/countly-storage.js [hunk 3]
```diff
@@ -101,11 +101,20 @@ const fileStorage = {
...
-        fs.unlink(dir, (err) => {
-            if (err) {
-                cc.log(cc.logLevelEnums.ERROR, `storeRemove, Failed to remove file with key: [${key}]. Error: [${err}].`);
+        var filePath = path.resolve(__dirname, `${getStoragePath()}__${key}.json`);
+        fs.access(filePath, fs.constants.F_OK, (accessErr) => {
+            if (accessErr) {
+                cc.log(cc.logLevelEnums.WARNING, `storeRemove, No file found with key: [${key}]. Nothing to remove.`);
+                return;
             }
+            fs.rmSync(filePath, { recursive: true, force: true }, (err) => {
```
Comment (turtledreams) [inline lib/countly-storage.js:110]: why a jump from unlink to rmSync?

**Review by turtledreams — CHANGES_REQUESTED** [review]

**Pull request merged**